tests/parametric/test_otel_span_methods.py

```diff
@@ -286,3 +286,608 @@ def test_otel_get_span_context(self, test_agent, test_library):
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_CLIENT) as span:
+                span.set_attributes({"db.system": "mongodb"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "mongodb.query"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "go", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_operation_name_message_consumer_001(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `messaging.system + "." + messaging.operation` when:
+            - Span kind is set to Client
+            - messaging.system is set to something
+            - messaging.operation is set to something
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_CLIENT) as span:
+                span.set_attributes({"messaging.system": "kafka"})
+                span.set_attributes({"messaging.operation": "receive"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "kafka.receive"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "go", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_operation_name_message_consumer_002(self, test_agent, test_library):
```

Comment:
@dougqh it looks like SpanKind.SERVER may also be used by messaging operations. What should the result be in this use case? I'm guessing the same `{messaging.system}.{messaging.operation}`?

Reply:
I mean in the DD libraries. Here are the possible SpanKind's and the expected (as of this writing) operation name:

| SpanKind | Operation Name |
|----------|----------------|
| CLIENT   | {messaging.system}.{messaging.operation} |
| PRODUCER | {messaging.system}.{messaging.operation} |
| CONSUMER | {messaging.system}.{messaging.operation} |
| SERVER   | _undefined_ |

We're currently missing a mapping for the SERVER span kind (which is [possible](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/#span-kind) for a messaging span) and I think it makes sense to apply the same convention to this case. Does this change make sense?

Reply:
I've re-arranged the file, but I've kept this `SERVER` case in, but it was missing from our spec document.

[Updated link to OpenTelemetry docs outlining that it can be a server](https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/#span-kind)